lib/ui/painting/codec.cc

```diff
@@ -36,6 +42,42 @@ enum PixelFormat {
   kBGRA8888,
 };
 
+#if OS_ANDROID
+
+// Compressed image buffers are allocated on the UI thread but are deleted on a
+// decoder worker thread.  Android's implementation of malloc appears to
+// continue growing the native heap size when the allocating thread is
+// different from the freeing thread.  To work around this, create an SkData
+// backed by an anonymous mapping.
+sk_sp<SkData> MakeSkDataWithCopy(const void* data, size_t length) {
+  size_t mapping_length = length + sizeof(size_t);
+  void* mapping = ::mmap(nullptr, mapping_length, PROT_READ | PROT_WRITE,
+                         MAP_ANONYMOUS | MAP_PRIVATE, -1, 0);
+
+  FML_CHECK(mapping != MAP_FAILED);
```

Comment:
I think we should return a nullptr SkData here to indicate OOM. We can check for a nullptr in the caller to return an error to Dart code.